telemetrylib/TCP.cpp

```diff
@@ -62,14 +70,40 @@ public slots:
         {
             QTcpSocket* sender = static_cast<QTcpSocket*>(QObject::sender());
             _sockets.removeOne(sender);
-            connected = false;
+            connection = false;
             emit connectionStatusChanged();
         }
     }
 private:
-    QTcpServer _server;     //the TCP socket
-    QList<QTcpSocket*> _sockets;    //The list of TCP socket
-    bool connected = false;     //This is used when getConnectionStatus() is called
+    QTcpServer _server;
+    QList<QTcpSocket*> _sockets;
+    std::atomic<bool> connection = false; // This is used when getConnectionStatus() is called
+    std::thread *t;
+    std::atomic<bool> finish = false; // For soft quiting the thread
+
+    /**
+     * Creates a thread that pings the engineering server's static IP to check connection
+     */
+    void checkConnection() {
+        QTcpSocket sock;
+        while (!finish) {
+            sock.connectToHost("192.168.1.16", 4005);
+            bool connected = sock.waitForConnected(500);
```

Comment:
Change wait time to 4000 from PR#17
```suggestion
            bool connected = sock.waitForConnected(4000);
```
```suggestion
            bool connected = sock.waitForConnected(500);
```

Reply:
Fixed